### PR TITLE
Possiblity to reassign problem components

### PR DESCRIPTION
**Pull request by marusak**

### CODE REVIEW

File: src/pyfaf/storage/migrations/versions/f43edd5b636d_enable_problem_components_reassign.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.create_table(
+        'problemreassign',
+        sa.Column('id', sa.Integer, primary_key=True),
+        sa.Column('date', sa.Date, nullable=False),
+        sa.Column('problem_id', sa.Integer, nullable=False),
```
Comment (xsuchy) [inline src/pyfaf/storage/migrations/versions/f43edd5b636d_enable_problem_components_reassign.py]: Can you please create index on this column? (hint: ForeignKey != Index)

File: src/webfaf/problems.py [hunk 2]
```diff
@@ -419,13 +424,54 @@ def dashboard():
...
     if problem is None:
         raise abort(404)
 
+    if component_names:
+        try:
+            current_components = (db.session.query(ProblemComponent)
+                                  .filter_by(problem_id=problem_id)
+                                  .all())
+            for c in current_components:
+                db.session.delete(c)
```
Comment (xsuchy) [inline src/webfaf/problems.py]: Can this be replaced by:
```
db.session.query(ProblemComponent).filter_by(problem_id=problem_id).delete()
```
?

File: src/webfaf/problems.py [hunk 3]
```diff
@@ -549,10 +596,18 @@ def item(problem_id):
...
+@problems.route("/bthash/<bthash>/", methods=["GET", "POST"])
 def bthash_forward(bthash=None):
     # single hash
+
+    # component ids can't be accessed through components_form object because of
+    # redirection, it must be passed to the item function as an parameter
+    if request.method == 'POST':
+        component_names = request.form.get('component_names')
+    else:
+        component_names = None
```
Comment (xsuchy) [inline src/webfaf/problems.py]: ?? Why you are ignoring the GET?

Comment (marusak) [reply src/webfaf/problems.py]: we are not ignoring GET, but only in POST there can come `component_names`.

**Comment by marusak** [comment]
@xsuchy I've updated the other issues

**Pull request merged**